src/dialog.lua

```diff
@@ -348,25 +348,10 @@ function DialogItemEdit:handleKeyPress(k, modifiers)
 end
 
 function DialogItemEdit:showEditor()
-    -- This is really nasty, we should do something better
-    local dlg = {
-        title = self.prompt,
-        items = {
-            {
-            type = dItemTypes.dEDIT,
-            prompt = self.prompt,
-            value = self.value,
-            },
-        },
-        buttons = {
-            { key = -27, text = "Cancel" },
-            { key = 13, text = "OK" },
-        },
-    }
-    if runtime:iohandler().dialog(dlg) > 0 then
-        self.value = dlg.items[1].value
-        self:setNeedsRedraw()
-    end
+    local trapped = runtime:getTrap()
```

Comment:
The getTrap/setTrap aren't needed here, that's only for opcodes that can be called with `TRAP foo` (and which I keep meaning to replace with a better impl because having individual opcodes remember to call setTrap really sucks).

As discussed, the readLine param should be better called allowCancel and it should always be `true` for the use we're making of it here.